src/epoch.rs

```diff
@@ -2226,6 +2242,218 @@ impl Epoch {
         me
     }
 
+    /// Converts this epoch into the time of week, represented as a rolling week counter into that time scale and the number of nanoseconds since closest Sunday midnight into that week.
+    /// This is usually how GNSS receivers describe a timestamp.
+    pub fn to_time_of_week(&self) -> (u32, u64) {
+        // fractional days in this time scale
+        let days = self.to_duration().to_unit(Unit::Day);
+        // wk: rolling week counter into timescale
+        let wk = (days / Weekday::DAYS_PER_WEEK).floor() as u32;
+        // tow: number of nanoseconds between self and previous sunday midnight / start of week
+        let mut start_of_week = self.previous_weekday_at_midnight(Weekday::Sunday);
+        // restrict start of week/sunday to the time scale starting day
+        // std::cmp::min( start_of_week, self.time_scale.ref_epoch)
+        match self.time_scale {
+            TimeScale::GPST => {
+                if start_of_week < GPST_REF_EPOCH {
+                    start_of_week = GPST_REF_EPOCH;
+                }
+            }
+            TimeScale::BDT => {
+                if start_of_week < BDT_REF_EPOCH {
+                    start_of_week = BDT_REF_EPOCH;
+                }
+            }
+            TimeScale::GST => {
+                if start_of_week < GST_REF_EPOCH {
+                    start_of_week = GST_REF_EPOCH;
+                }
+            }
+            TimeScale::TAI => {
+                if start_of_week < J1900_REF_EPOCH {
+                    start_of_week = J1900_REF_EPOCH;
+                }
+            }
+            _ => {}
+        }
```

Comment:
@ChristopherRabotin,
the previous implementation behaved funny for extreme values (inside the first week of a given timescale).
Because a timescale possibly did not start on a monday, and `previous_weekday_at_midnight` would determine a day prior `REF_EPOCH` . It makes more sense to restrict to the timescale, since these rolling counters are usually implemented in the form of unsigned integer. `u16` is truly how a GPS vehicle count the elapsed weeks for instance.

With this, i'm able to get the tests to pass for extreme values, see the testbench.

Also, i don't unterstand why the UNIX timescale is not wrapped in our timescales, it could be convenient for it to benefit all the existing methods, not only these ones. 

I will try to get the reciprocal test to work today